### PR TITLE
Fix metrics asg

### PR DESCRIPTION
**Pull request by multani**
#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### Purpose

This makes the metrics check work at last :)

#### Known Compatibility Issues

The ``--statistic`` flag has been removed but it never worked in the first place.
``--scheme`` wasn't doing anything before but it now does.

### CODE REVIEW

File: bin/metrics-asg.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #
 # NOTES:
 #   Returns latency statistics by default.  You can specify any valid ASG metric type, see
-#   http://http://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/as-metricscollected.html
+#   https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/as-metricscollected.html
```
Comment (majormoses) [inline bin/metrics-asg.rb]: :+1:

File: bin/metrics-asg.rb [hunk 2]
```diff
@@ -101,7 +94,7 @@ def cloud_watch_metric(metric_name, value, asg_name)
       metric_name: metric_name,
       dimensions: [
         {
-          name: 'AutoScaling',
+          name: 'AutoScalingGroupName',
```
Comment (majormoses) [inline bin/metrics-asg.rb]: good catch.

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

File: bin/metrics-asg.rb [hunk 3]
```diff
@@ -114,33 +107,34 @@ def cloud_watch_metric(metric_name, value, asg_name)
...
-      result['AutoScalingGroup.' + asg_name + '.' + key + '.' + static] = r[:datapoints][0] unless r[:datapoints][0].nil?
+      keys =
+        if config[:scheme].nil?
+          []
+        else
+          [config[:scheme]]
+        end
+      keys.concat ['AutoScalingGroup', asg_name, key, static]
+
+      result[keys.join('.')] = r[:datapoints][0] unless r[:datapoints][0].nil?
```
Comment (majormoses) [inline bin/metrics-asg.rb]: I personally prefer `.first` over `[0]` as it is a bit more expressive but I certainly won't make you change it if you don't want to.

File: bin/metrics-asg.rb [hunk 4]
```diff
@@ -61,12 +60,6 @@ class ASGMetrics < Sensu::Plugin::Metric::CLI::Graphite
          long: '--metric',
          default: 'GroupInServiceInstances'
 
-  option :statistic,
-         description: 'Statistics type',
-         short: '-t STATISTIC',
-         long: '--statistic',
-         default: ''
```
Comment (majormoses) [inline bin/metrics-asg.rb]: why not default this to `Sum` and then it can be fixed to actually work?

File: bin/metrics-asg.rb [hunk 5]
```diff
@@ -114,33 +107,34 @@ def cloud_watch_metric(metric_name, value, asg_name)
...
-        'GroupStandbyInstances' => 'Sum',
-        'GroupTerminatingInstances' => 'Sum',
-        'GroupTotalInstances' => 'Sum'
-      }
-      statistic = default_statistic_per_metric
-    else
-      statistic = config[:statistic]
-    end
+    statistic = {
+      'GroupMinSize' => 'Sum',
```
Comment (majormoses) [inline bin/metrics-asg.rb]: If you default the `statistic`  option to `Sum` you can use:
```
statistic = {
  'GroupMaxSize' => config[:statistic],
  ...
}
```

**Comment by majormoses** [comment]
Also can I get a [testing artifact](https://github.com/sensu-plugins/community/blob/master/PULL_REQUEST_PROCESS.md#7-testing-artifacts)

**Comment by multani** [comment]
I've just pushed the changes requested, I'm perfectly fine with ``.first`` instead of ``[0]``, and your proposition for the ``--statistic`` flag makes lot of sense, I don't know why I didn't think about it 🤦‍

I actually commited a changelog entry ... but didn't push it 😅 So here you go.

I also added a new change to the original PR to properly "escape" autoscaling group names containing spaces, otherwise we won't be able to measure half of our groups actually 😅

Finally, here's the test artifact (redacted):

Before:
```
$ bundle exec bin/metrics-asg.rb
$ bundle exec bin/metrics-asg.rb --scheme foo
$ bundle exec bin/metrics-asg.rb --scheme foo --statistic Average
Check failed to run: undefined method `each' for "Average":String, ["bin/metrics-asg.rb:118:in `print_statistics'", "bin/metrics-asg.rb:151:in `run'", "/home/jballet/projects/sensu/plugins/sensu-plugins-aws/.gems/ruby/2.3.0/gems/sensu-plugin-2.3.0/lib/sensu-plugin/cli.rb:57:in `block in <class:CLI>'"]
$ bundle exec bin/metrics-asg.rb --name "Name with space"
```

After:
```
$ bundle exec bin/metrics-asg.rb
autoscalinggroup.xxx.groupminsize.sum 3.0 1512291600
autoscalinggroup.xxx.groupmaxsize.sum 7.0 1512291600
autoscalinggroup.xxx.groupdesiredcapacity.sum 5.0 1512291600
autoscalinggroup.xxx.groupinserviceinstances.sum 5.0 1512291600
autoscalinggroup.xxx.grouppendinginstances.sum 0.0 1512291600
autoscalinggroup.xxx.groupstandbyinstances.sum 0.0 1512291600
autoscalinggroup.xxx.groupterminatinginstances.sum 0.0 1512291600
autoscalinggroup.xxx.grouptotalinstances.sum 5.0 1512291600
$ bundle exec bin/metrics-asg.rb --scheme foo
foo.autoscalinggroup.xxx.groupminsize.sum 3.0 1512291600
foo.autoscalinggroup.xxx.groupmaxsize.sum 7.0 1512291600
foo.autoscalinggroup.xxx.groupdesiredcapacity.sum 5.0 1512291600
foo.autoscalinggroup.xxx.groupinserviceinstances.sum 5.0 1512291600
foo.autoscalinggroup.xxx.grouppendinginstances.sum 0.0 1512291600
foo.autoscalinggroup.xxx.groupstandbyinstances.sum 0.0 1512291600
foo.autoscalinggroup.xxx.groupterminatinginstances.sum 0.0 1512291600
foo.autoscalinggroup.xxx.grouptotalinstances.sum 5.0 1512291600
$ bundle exec bin/metrics-asg.rb --scheme foo --statistic Average
foo.autoscalinggroup.xxx.groupminsize.average 3.0 1512291660
foo.autoscalinggroup.xxx.groupmaxsize.average 7.0 1512291660
foo.autoscalinggroup.xxx.groupdesiredcapacity.average 5.0 1512291660
foo.autoscalinggroup.xxx.groupinserviceinstances.average 5.0 1512291660
foo.autoscalinggroup.xxx.grouppendinginstances.average 0.0 1512291660
foo.autoscalinggroup.xxx.groupstandbyinstances.average 0.0 1512291660
foo.autoscalinggroup.xxx.groupterminatinginstances.average 0.0 1512291660
foo.autoscalinggroup.xxx.grouptotalinstances.average 5.0 1512291660
$ bundle exec bin/metrics-asg.rb --name "Name with space"
autoscalinggroup.name_with_space.groupminsize.sum 1.0 1512292140
autoscalinggroup.name_with_space.groupmaxsize.sum 10.0 1512292140
autoscalinggroup.name_with_space.groupdesiredcapacity.sum 3.0 1512292140
autoscalinggroup.name_with_space.groupinserviceinstances.sum 3.0 1512292140
autoscalinggroup.name_with_space.grouppendinginstances.sum 0.0 1512292140
autoscalinggroup.name_with_space.groupstandbyinstances.sum 0.0 1512292140
autoscalinggroup.name_with_space.groupterminatinginstances.sum 0.0 1512292140
autoscalinggroup.name_with_space.grouptotalinstances.sum 3.0 1512292140
```

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-aws/versions/10.0.3

**Pull request merged**